src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const i of action.keysToRemove) {
+          if (stateCopy[i]) {
+            delete stateCopy[i];
```

Comment:
The check `if (stateCopy[i])` is not necessary and can lead to incorrect behavior if a property exists but has a falsy value (e.g., `null`, `0`, `''`, `false`). You should delete the property without checking its value since the task specifies to ignore non-existing keys.